### ISSUE TITLE
IndexOutOfRangeException for obfuscated assemblies

### ISSUE THREAD

**Issue by EFLFE** [jbevain/cecil#482]
An error appears when open the obfuscated assembly. This may be logical for a secure assembly, but because of this error breaks the ILSpy ([when searching](https://github.com/icsharpcode/ILSpy/issues/1031)).

![img](https://camo.githubusercontent.com/9e3476b3eb850aea5099ab266cf9da88cc02650e/68747470733a2f2f692e696d6775722e636f6d2f35344c444f41582e706e67)

What I did: I used the ConfuserEx with one rule: "invalid metadata" (This protection adds invalid metadata to modules to prevent disassembler/decompiler from opening them)

Looks like this:

![img](https://camo.githubusercontent.com/f5e6d1646efbe5fd1bb17543ce3661e69a1d4f83/68747470733a2f2f692e696d6775722e636f6d2f6f30737237384e2e706e67)

Obfuscated assembly: [Lidgren.Network.dll.zip](https://github.com/jbevain/cecil/files/1638254/Lidgren.Network.dll.zip)

**Comment by bprg** [comment]
cecil doesn't support obfuscated assemblies.

**Comment by jbevain** [comment]
Yeah that assembly is completely broken. I suggest ILSpy protects against that because it seems like we can throw from a thousand places with an assembly like that.
